### PR TITLE
Refactor according to n8n PR comments 

### PR DESCRIPTION
**Pull request by YokySantiago**
Some changes have been added according to n8n PR comments so, now we have:

- FixedCollection for destination and recipient which are collections used by other entities 
- Loading information from Onfleet API like: Teams, Hubs, Workers, Admins and Time zones

Also, the code is formatted with nodelinter.

### CODE REVIEW

File: packages/nodes-base/nodes/Onfleet/GenericFunctions.ts [hunk 1]
```diff
@@ -40,17 +42,63 @@ export async function onfleetApiRequest(
...
-			const { message = '', cause = '' } = messageError as IDataObject;
-			if (message && cause) {
-				apiError.message = `${message}: ${cause}`;
-			} else {
-				apiError.message = message;
-			}
-			throw new NodeApiError(this.getNode(), apiError as JsonObject);
-		}
-		throw new NodeApiError(this.getNode(), apiError as JsonObject);
+		throw new NodeApiError(this.getNode(), error as JsonObject); // TODO: Check error
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/GenericFunctions.ts]: Do we intend to keep the todo?

File: packages/nodes-base/nodes/Onfleet/Onfleet.node.ts [hunk 2]
```diff
@@ -271,22 +366,21 @@ export class Onfleet implements INodeType {
...
 			/* -------------------------------------------------------------------------- */
 			const name = this.getNodeParameter('name', item) as string;
 			const phone = this.getNodeParameter('phone', item) as string;
-			const teams = JSON.parse(this.getNodeParameter('teams', item) as string) as string[];
-			const vehicle = this.getNodeParameter('vehicle', item) as boolean;
+			const teams = this.getNodeParameter('teams', item) as string[];
+			const { vehicleProperties: vehicle } = this.getNodeParameter('vehicle', item) as IDataObject;
 			const workerData: OnfleetWorker = { name, phone, teams };
 
 			// Addding vehicule fields
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/Onfleet.node.ts]: Typos

File: packages/nodes-base/nodes/Onfleet/descriptions/WebhookDescription.ts [hunk 3]
```diff
@@ -20,26 +20,26 @@ export const webhookOperations = [
...
-] as INodeProperties[];
+];
 
 const urlField = {
 	displayName: 'Url',
 	name: 'url',
 	type: 'string',
 	default: '',
-	description: 'The URL that Onfleet should issue a request against as soon as the trigger condition is met. It must be HTTPS and have a valid certificate',
+	description: 'The URL that Onfleet should issue a request against as soon as the trigger condition is met. It must be HTTPS and have a valid certificate.',
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/WebhookDescription.ts:42]: I thought we are removing periods at the end of the descriptions ;)

File: packages/nodes-base/nodes/Onfleet/descriptions/TaskDescription.ts [hunk 4]
```diff
@@ -365,16 +423,16 @@ export const taskFields = [
 				displayName: 'Recipient Skip SMS Notifications Override',
 				name: 'recipientSkipSMSNotifications',
 				type: 'boolean',
-				default: '',
-				description: 'Override the recipient notification settings for this task only',
+				default: false,
+				description: 'Whether override the recipient notification settings for this task only or not',
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/TaskDescription.ts]: `Whether to override the recipient notification settings for this task` is more like what we want to describe here

File: packages/nodes-base/nodes/Onfleet/descriptions/TaskDescription.ts [hunk 5]
```diff
@@ -365,16 +423,16 @@ export const taskFields = [
...
 			},
 			{
 				displayName: 'Use Merchant For Proxy Override',
 				name: 'useMerchantForProxy',
 				type: 'boolean',
-				default: '',
-				description: 'Override the organization ID to use the merchant orgID when set to true for this task only',
+				default: false,
+				description: 'Whether override the organization ID to use the merchant orgID when set to true for this task only',
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/TaskDescription.ts]: `Whether to override the organization ID with the merchant's org ID for this task` is more like what we want to describe here

File: packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts [hunk 6]
```diff
@@ -2,56 +2,56 @@ import {
...
 			show: {
-				resource: [ 'teams' ],
+				resource: [ 'team' ],
 			},
 		},
 		options: [
+			{
+				name: 'Auto-Dispatch',
+				value: 'autoDispatch',
+				description: 'Dynamically dispatching tasks on the fly',
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts]: This isn't a good description of the feature, rename to `Automatically dispatch tasks assigned to a team to on-duty drivers`

File: packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts [hunk 7]
```diff
@@ -2,56 +2,56 @@ import {
...
 			{
 				name: 'Get',
 				value: 'get',
 				description: 'Get a specific Onfleet team',
 			},
 			{
-				name: 'List',
+				name: 'Get All',
 				value: 'getAll',
 				description: 'List all Onfleet teams',
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts]: Should be Get all as we're unifying this

File: packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts [hunk 8]
```diff
@@ -2,56 +2,56 @@ import {
...
 			{
-				name: 'Auto-Dispatch',
-				value: 'autoDispatch',
-				description: 'Dynamically dispatching tasks on the fly',
-			},
-			{
-				name: 'Get time estimates',
+				name: 'Get Time Estimates',
 				value: 'getTimeEstimates',
 				description: 'The Driver Time Estimates endpoint allows an API user to get estimated times for tasks that haven\'t been created yet',
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts]: `Get estimated times for upcoming tasks for a team, returns a selected driver`

File: packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts [hunk 9]
```diff
@@ -62,39 +62,48 @@ const nameField = {
...
 } as INodeProperties;
 
 const enableSelfAssignmentField = {
-	displayName: 'Self assignment',
+	displayName: 'Self Assignment',
 	name: 'enableSelfAssignment',
 	type: 'boolean',
 	default: false,
-	description: 'This toggles Team Self-Assignment that allows Drivers to Self Assign Tasks that are in the Team unassigned container',
+	description: 'Whether allows Drivers to Self Assign Tasks that are in the Team unassigned container or not',
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts]: `Whether or not to allow drivers to self-assign tasks that are in the Team's unassigned container`

File: packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts [hunk 10]
```diff
@@ -133,94 +126,78 @@ const serviceTimeField = {
...
 } as INodeProperties;
 
 const pickupTimeField = {
-	displayName: 'Pickup time',
+	displayName: 'Pick Up Time',
 	name: 'pickupTime',
 	type: 'dateTime',
-	default: Date.now(),
+	default: '',
 	description: 'If the request includes pickupLocation pickupTime must be present if the time is fewer than 3 hours in the future',
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts]: add a `,` between pickupLocation and pickupTime

File: packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts [hunk 11]
```diff
@@ -246,7 +223,7 @@ const restrictedVehicleTypesField = {
 } as INodeProperties;
 
 const serviceTimeEstimateField = {
-	displayName: 'Service time',
+	displayName: 'Service Time',
 	name: 'serviceTIme',
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts]: `serviceTime`

File: packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts [hunk 12]
```diff
@@ -345,109 +322,170 @@ export const teamFields = [
...
 			},
-		},
+		],
 	},
 	{
-		displayName: 'Additional fields',
+		displayName: 'Additional Fields',
 		name: 'additionalFields',
 		type: 'collection',
 		placeholder: 'Add fields',
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts:472]: I think this should be singular as we do in other places

File: packages/nodes-base/nodes/Onfleet/descriptions/WorkerDescription.ts [hunk 13]
```diff
@@ -112,16 +110,12 @@ const vehicleTypeField = {
...
-			value: 'BICYCLE',
-		},
 		{
 			name: 'Truck',
 			value: 'TRUCK',
 		},
 	],
-	default: 'CAR',
+	default: '',
 	description: 'Whether the worker has vehicle or not',
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/WorkerDescription.ts]: Best to clarify if not provided, this worker will be treated as if on foot.

File: packages/nodes-base/nodes/Onfleet/descriptions/WorkerDescription.ts [hunk 14]
```diff
@@ -191,16 +191,102 @@ const phonesFilterField = {
...
+			value: 'vehicle',
+		},
+		{
+			name: 'Worker ID',
+			value: 'id',
+		},
+	],
 	default: '',
-	description: 'A comma-separated list of fields to return, if all are not desired. For example, name, location',
+	description: 'A list of fields to return, if all are not desired',
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/WorkerDescription.ts]: I don't get this description, can we rephrase it to something more explanatory?

File: packages/nodes-base/nodes/Onfleet/descriptions/WorkerDescription.ts [hunk 15]
```diff
@@ -234,21 +320,24 @@ const radiusFilterField = {
 		minValue: 0,
 	},
 	default: 1000,
-	description: 'The length in meters of the radius of the spherical area in which to look for workers. Defaults to 1000 if missing. Maximum value is 10000',
+	description: 'The length in meters of the radius of the spherical area in which to look for workers. Defaults to 1000 if missing. Maximum value is 10000.',
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/WorkerDescription.ts:323]: I think the consensus is to not include the period at the end of descriptions?

File: packages/nodes-base/nodes/Onfleet/descriptions/RecipientDescription.ts [hunk 16]
```diff
@@ -229,71 +238,39 @@ export const recipientFields = [
...
 		options: additionalRecipientFields,
 	},
 	{
-		displayName: 'Additional recipient fields',
-		name: 'additionalRecipientFields',
+		displayName: 'Update Fields',
+		name: 'updateFields',
 		type: 'collection',
-		placeholder: 'Add recipient fields',
+		placeholder: 'Update fields',
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/RecipientDescription.ts]: Do we capitalize the `F` and also singular?

File: packages/nodes-base/nodes/Onfleet/descriptions/RecipientDescription.ts [hunk 17]
```diff
@@ -58,53 +58,93 @@ const additionalRecipientFields = [
...
+	default: '',
 } as INodeProperties;
 
 const recipientPhone = {
-	displayName: 'Recipient phone',
+	displayName: 'Recipient Phone',
 	name: 'recipientPhone',
 	type: 'string',
-	description: 'A unique, valid phone number as per the organization\'s country if there\'s no leading + sign. If a phone number has a leading + sign, it will disregard the organization\'s country setting',
+	description: 'A unique, valid phone number as per the organization\'s country if there\'s no leading + sign. If a phone number has a leading + sign, it will disregard the organization\'s country setting.',
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/RecipientDescription.ts:72]: No period?

File: packages/nodes-base/nodes/Onfleet/descriptions/OrganizationDescription.ts [hunk 18]
```diff
@@ -2,46 +2,46 @@ import {
...
 		default: 'get',
 	},
-] as INodeProperties[];
+];
 
-export const organizationFields = [
+export const organizationFields: INodeProperties[] = [
 	{
-		displayName: 'ID',
+		displayName: 'Organiation ID',
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/OrganizationDescription.ts]: Organization typo

File: packages/nodes-base/nodes/Onfleet/descriptions/DestinationDescription.ts [hunk 19]
```diff
@@ -27,22 +27,22 @@ export const destinationOperations = [
...
 	},
-] as INodeProperties[];
+];
 
 const unparsedField = {
-	displayName: 'Unparsed adress',
+	displayName: 'Unparsed Address',
 	name: 'unparsed',
 	type: 'boolean',
 	description: 'Whether the address is specified in a single',
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/descriptions/DestinationDescription.ts]: Unfinished sentence here: `Whether or not the address is specified in a single unparsed string`

File: packages/nodes-base/nodes/Onfleet/Onfleet.node.ts [hunk 20]
```diff
@@ -67,124 +78,211 @@ export class Onfleet implements INodeType {
...
+				try {
+					await this.helpers.request(options);
+					return {
+						status: 'OK',
+						message: 'Authentication successful',
+					};
+				} catch (error) {
+					return {
+						status: 'Error',
+						message: `Settings are not valid: ${error}`,
```
Comment (0xsemaj) [inline packages/nodes-base/nodes/Onfleet/Onfleet.node.ts]: I would specify Auth settings are not valid

**Review by jamesliupenn — CHANGES_REQUESTED** [review]

**Comment by 0xsemaj** [comment]
@YokySantiago please check the latest notes and make the changes accordingly, otherwise, everything else looks good to me!

File: packages/nodes-base/nodes/Onfleet/descriptions/WebhookDescription.ts (hunk 3 above)
Comment (YokySantiago) [reply packages/nodes-base/nodes/Onfleet/descriptions/WebhookDescription.ts:42]: Yes, you're right but according to the node linter given by n8n, multi-paragraphs must end with dot

File: packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts (hunk 12 above)
Comment (YokySantiago) [reply packages/nodes-base/nodes/Onfleet/descriptions/TeamDescription.ts:472]: According to Ricardo from n8n and the node linter it can be `Additional Fields`

File: packages/nodes-base/nodes/Onfleet/descriptions/WorkerDescription.ts (hunk 15 above)
Comment (YokySantiago) [reply packages/nodes-base/nodes/Onfleet/descriptions/WorkerDescription.ts:323]: Yes, you're right but according to the node linter given by n8n, multi-paragraphs must end with dot

File: packages/nodes-base/nodes/Onfleet/descriptions/RecipientDescription.ts (hunk 16 above)
Comment (YokySantiago) [reply packages/nodes-base/nodes/Onfleet/descriptions/RecipientDescription.ts]: You're right, we need to capitalize the F but the name of the field is right according to node linter

File: packages/nodes-base/nodes/Onfleet/descriptions/RecipientDescription.ts (hunk 17 above)
Comment (YokySantiago) [reply packages/nodes-base/nodes/Onfleet/descriptions/RecipientDescription.ts:72]: Yes, you're right but according to the node linter given by n8n, multi-paragraphs must end with dot

**Comment by 0xsemaj** [comment]
Lgtm, merging

**Review by jamesliupenn — COMMENTED** [review]
Not sure what's going on here but the review wasn't properly closed, re-opening it so I can close it

**Comment by 0xsemaj** [comment]
Well, my checkmark disappeared and I cannot locate to actually approve this on Github......
